src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -83,42 +83,28 @@ struct Cone{T,CO,TR,TP<:Union{Nothing,T}} <: AbstractVolumePrimitive{T, CO}
 end
 
 #Type conversion happens here
-function Cone{T,CO}(r, φ::Tuple{T,T}, hZ, origin, rotation) where {T,CO}
-    r = _csg_convert_args(T, r)
-    φ = _csg_convert_args(T, φ)
-    _φ = abs(φ[2]-φ[1])
-    rotation=rotation*SMatrix{3}(cos(φ[1]),sin(φ[1]),0,-sin(φ[1]),cos(φ[1]),0,0,0,1)
-    hZ = _csg_convert_args(T, hZ)
-    Cone{T,CO,typeof(r),typeof(_φ)}(r, _φ, hZ, origin, rotation)
-end
-
 function Cone{T,CO}(r, φ, hZ, origin, rotation) where {T,CO}
     r = _csg_convert_args(T, r)
     φ = _csg_convert_args(T, φ)
+    (_φ, rotation) = _handle_phi(φ, rotation)
     hZ = _csg_convert_args(T, hZ)
-    Cone{T,CO,typeof(r),typeof(φ)}(r, φ, hZ, origin, rotation)
+    Cone{T,CO,typeof(r),typeof(_φ)}(r, _φ, hZ, origin, rotation)
 end
```

Comment:
@fhagemann already commented using the same names (`r = _csg_convert_args(T, r); ...`) could create some type instabilities. I tested it and it does. You can see it yourself with `@code_warntype Cone{Float64,CO}(1, ...)`. 

We have to use different names for the variables, e.g. `_r = _csg_convert_args(T, r); ...`.